boards: add WeAct Studio ESP32-S3-B board support

heronet opened the pull request:
Add support for WeAct Studio ESP32-S3-B development board featuring:
- ESP32-S3R8 chip with 8MB integrated PSRAM
- 16MB external SPI flash
- Dual USB-C connectors
- Wi-Fi and Bluetooth LE support
- RGB WS2812 LED on GPIO48
- User button on GPIO45

heronet commented on boards/weact/weact_esp32s3_b/weact_esp32s3_b_procpu.dts:
```diff
@@ -0,0 +1,188 @@
+
+	buttons {
+		compatible = "gpio-keys";
+		button0: button_0 {
```
fixed throughout the pr

nordicjm commented on boards/weact/weact_esp32s3_b/weact_esp32s3_b_procpu.dts:
```diff
@@ -0,0 +1,189 @@
+	pinctrl-names = "default";
+};
+
+&i2s0_default {
+	group1 {
+		pinmux = <I2S0_O_SD_GPIO48>;
+		output-enable;
+	};
+};
```
this should be in pinctrl file

heronet committed:
boards: add WeAct Studio ESP32-S3-B board support

Add support for WeAct Studio ESP32-S3-B development board featuring:
- ESP32-S3R8 chip with 8MB integrated PSRAM
- 16MB external SPI flash
- RGB WS2812 LED on GPIO48
- User button on GPIO45
- Dual USB-C connectors
- Wi-Fi and Bluetooth LE support

Signed-off-by: Siratul Islam <sirat4757@gmail.com>

heronet commented:
Hi @kartben, @nordicjm,

Just wanted to check in regarding this PR. Is there anything else I should address, or any feedback needed to help move it forward? Please let me know if any changes are required.

Thank you for your time and reviewing!
